src/main/java/frc/robot/InitBindings.java

```diff
@@ -155,6 +177,32 @@ public static void initDriveBindings(Drive drive, StrategyManager strategyManage
                       strategyManager.setAutonomyMode(AutonomyMode.Mixed);
                       // And then fall through to the mixed autonomy behavior (no break here is
                       // intentional)
+                    case Smart:
+                      // If in algae mode, automatically set level and pick nearest algae location
+                      if (ScoringSubsystem.getInstance() == null
+                          || ScoringSubsystem.getInstance().getGamePiece() == GamePiece.Algae) {
+                        DesiredLocation desiredLocation =
+                            ReefLineupUtil.getClosestAlgaeLocation(drive.getPose());
+                        drive.setDesiredLocation(desiredLocation);
+
+                        // Set algae level automatically
+                        if (ScoringSubsystem.getInstance() != null) {
+                          ScoringSubsystem.getInstance()
+                              .setTarget(
+                                  ReefLineupUtil.getAlgaeLevelFromDesiredLocation(desiredLocation));
+                        }
+                      } else {
+                        if (DriverStation.getAlliance().isPresent()
+                            && DriverStation.getAlliance().get() == Alliance.Red) {
+                          drive.setDesiredLocation(
+                              JsonConstants.redFieldLocations.getClosestCoralStation(
+                                  drive.getPose()));
+                        } else {
+                          drive.setDesiredLocation(
```

Comment:
This and other drive classes will NPE if we feature flag out drive right? Maybe we need to clean that up here

Reply:
Oh okay cool - I think I just lost that context from reviewing on my phone